File: Serene/Serene.Script/Imports/FormContexts/Membership.LoginForm.cs

namespace Serene.Membership
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class LoginForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Membership.Login";
    
        public LoginForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor Username { [InlineCode("{this}.w('Username', Serenity.StringEditor)")] get; private set; }
        public PasswordEditor Password { [InlineCode("{this}.w('Password', Serenity.PasswordEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Membership.LoginForm.ts
namespace Serene.Membership {
    export class LoginForm extends Serenity.PrefixedContext {
        static formKey = 'Membership.Login';

    }

    export interface LoginForm extends Serenity.PrefixedContext {
        Username(): Serenity.StringEditor;
        Password(): Serenity.PasswordEditor;
    }

    [['Username', Serenity.StringEditor], ['Password', Serenity.PasswordEditor]].forEach(x => LoginForm.prototype[<string>x[0]] = function() { return this.w(x[0], x[1]); });
}

